go/mcap/writer.go

```diff
@@ -607,10 +607,21 @@ func (w *Writer) Close() error {
 			}
 		}
 	}
+	var summaryCrc uint32
+	if summarySectionStart != 0 && w.opts.IncludeCRC {
+		// add partial footer record to end of summary section CRC
+		footerPrefix := make([]byte, 1+8+8+8)
+		footerPrefix[0] = byte(OpFooter)
+		putUint64(footerPrefix[1:], 8+8+4)
+		putUint64(footerPrefix[1+8:], summarySectionStart)
+		putUint64(footerPrefix[1+8+8:], summaryOffsetStart)
+		w.w.w.crc.Write(footerPrefix)
+		summaryCrc = w.w.Checksum()
+	}
```

Comment:
I think this can be replaced with something like,
```
err = w.WriteFooter(&Footer{
  SummarySectionStart:...,
  SummarySectionOffset: ...,
  CRC: ...,
})
```

Reply:
oh, sorry - my mistake. You are making this call below. I see what you are doing..

Reply:
what do you think of the idea of WriterFooter computing this CRC, while making an assumption that the CRC has been cleared after completion of the data section (after DataEnd, which I think is required?) Then we wouldn't need to do this partial footer record business, I believe

Reply:
a CRC provided to WriterFooter could be ignored in favor of the computation